crates/matrix-sdk/src/room/mod.rs

```diff
@@ -2360,6 +2360,39 @@ impl Room {
         // Get the user-defined mode if available
         notification_settings.get_user_defined_room_notification_mode(self.room_id()).await
     }
+
+    /// Report an event as inappropriate to the homeserver's administrator.
+    ///
+    /// # Arguments
+    ///
+    /// * `event_id` - The ID of the event to report.
+    /// * `score` - The score to rate this content, where `-100` is most
+    ///   offensive and `0` is inoffensive.
```

Comment:
Is the score server-checked? Are the bounds defined per the spec, or more of an informal standard by the community? If the former, could we check the score is in bounds, in this method?

Reply:
I don't know if the score is checked by the server, but the bounds are defined in the spec.

Reply:
In that case, would it make sense to check the bounds client-side? (and if so, add a test for it)

Reply:
I am wondering if we should declare a new type just for that argument rather than using `Int`. That way it can have its own error type while implementing `TryFrom` without having to add a variant to the `Error` enum just for this method.

Reply:
Indeed, that would be sweet!

Reply:
Done :slightly_smiling_face: